main_with_gui.py

```diff
@@ -154,6 +154,16 @@ def prompt_gpu_location(self, window: QMainWindow):
 
 	def generate_files(self, window: QMainWindow, gpu_loc: GPU):
 		try:
+			check_dep = sp.getstatusoutput([sys.executable, 'check_dependencies.sh'])
+			check_dep = check_dep[0]
+			if check_dep == 1:
+				buttonReply = QMessageBox.question(self, 'Install dependencies',
+												   "There aren't all the dependencies needed; do you want to install them?",
+												   QMessageBox.Yes | QMessageBox.No, QMessageBox.No)
+				if buttonReply == QMessageBox.Yes:
+					sp.call([sys.executable, "check_dependencies.sh"])
```

Comment:
Shouldn't this install the dependencies instead of checking for them again?
```suggestion
					sp.call([sys.executable, "./install_dependencies_all.sh"])
```
Please check if root permissions are needed, in that case you can't use `sudo` directly, but you have to use the `pkexec` solution already used in `main_with_gui.py` when running `generate_files.sh`

Reply:
We have discussed it now, resolving